builder/make_list_build_artifacts

```diff
@@ -11,8 +11,8 @@ artifacts=(".build/$cname-$COMMIT.tar" ".build/$cname-$COMMIT.release" ".build/$
 
 for feature in "${features[@]}"; do
 	for i in "features/$feature/"{image,convert}.*; do
-		extension="$(grep -E -o '(\.[a-z][a-zA-Z0-9\-_]*)*$' <<< "$i")"
-		artifacts+=(".build/$cname-$COMMIT$extension")
```

Comment:
When I see this right, we have this part of the code [duplicated](https://github.com/gardenlinux/builder/pull/97/files#diff-ba895b3419ad7b4b606b92528bdc561a3e240500b4c7a351ec8f8a748fa6c4e3R10) within this project? Not only is this code difficult to read/understand for someone who doesn't know the regex by hard. It also adds the issue that it is challenging to maintain.

Please put it into a function that can be called with some documentation to it. Since you've changed the extension that is included, this should be part of the documentation as well. I couldn't tell why we suddenly needed the tilde.